jerry-math/jerry-math-internal.h

```diff
@@ -35,7 +35,11 @@
  * If byte order is supported and set to little or target is among common
  * cases checked define __LITTLE_ENDIAN.
  */
-#if (defined (i386) || defined (__i386) || defined (__i386__) || \
+#if defined (_MSC_VER)
+#if defined (_M_IX86) || defined (_M_AMD64) || defined (_M_ARM) || defined (_M_ARM64)
+#define __LITTLE_ENDIAN
+#endif
```

Comment:
Don't do this. Just add the `defined (_M_IX86) || defined (_M_AMD64) || defined (_M_ARM) || defined (_M_ARM64)` line to the existing list. BTW, you've omitted `_M_X64` from the original suggestion.

It doesn't matter if some macros are only defined by gcc or clang, others only by msvc. They can be tested in the same list. They are just macros. It makes no harm if a compiler doesn't define / use / recognize some of them (as long as they are tested with `defined()`.